src/main/java/com/artipie/settings/StorageAliases.java

```diff
@@ -32,14 +35,38 @@ public interface StorageAliases {
     String FILE_NAME = "_storages.yaml";
 
     /**
-     * Find storage by alias.
+     * Finds or create storage by alias.
+     *
+     * @param cache Storages cache
      * @param alias Storage alias
      * @return Storage instance
      */
-    Storage storage(String alias);
+    Storage storage(StoragesCache cache, String alias);
+
+    /**
+     * Finds or create storage by config.
+     *
+     * @param cache Storages cache
+     * @param node Storage config
+     * @return Storage instance
+     */
+    default Storage storage(StoragesCache cache, YamlNode node) {
```

Comment:
It does not seem like `StorageAliases` actually need to be interface... Maybe we can convert it to class with empty ctor? In this case it seems like we can get rid of the `StorageAliases` field in several classes...